lib/checks/isPolynomialToConstantPower.js

```diff
@@ -0,0 +1,19 @@
+const Node = require('../node');
+
+function isPolynomialToConstantPower(node) {
+  // For the special case when the node is in the form (x + 2)^2
+  // First check if is power node and the exponent is a constant
+  // Then check if the base is a polynomial
+
+  if (!(Node.Type.isOperator(node, '^')
+        && Node.Type.isParenthesis(node.args[0])
+        && Node.Type.isConstant(node.args[1]))) {
+    return false;
+  }
+  const base = node.args[0].content;
+  const isPolynomial = base.args.some(Node.PolynomialTerm.isPolynomialTerm);
```

Comment:
Yeah, I think checking for isOperator(+) would be good and also address the issue I was about to raise of "how do you know `base` has args"?